README.md
#SoulNet
SoulBack.Net Code

# Website Changelog


## Website Update 0.1.4 - 11.12.2023
- Neues Dokument hochgeladen
- Download Page etwas übersichtlicher gestaltet


## Website Update 0.1.3 - 10.12.2023
Automated Github Action eingerichtet, für schnellen und automatisches Deploying auf den Webspace


## Website Update 0.1.2 - 10.12.2023
Routing fixed mittels searchParametern und meta-Refresh-Weiterleitung von /de/hoga


## Website Update 0.1.1 - 09.12.2023
Komplettes Backend der Website überarbeitet. Nun bessere und schnellere Funktion und bessere Entwicklung möglich. Beim nächsten Update hoffentlich Versionskontrolle auf Github inkl. automatischem Deploy auf den Webspace


## Website Update 0.1.0 - 08.12.2023
Website ging online


public/changelog.md
# Website Changelog


## Website Update 0.1.4 - 11.12.2023
- Neues Dokument hochgeladen
- Download Page etwas übersichtlicher gestaltet


## Website Update 0.1.3 - 10.12.2023
Automated Github Action eingerichtet, für schnellen und automatisches Deploying auf den Webspace


## Website Update 0.1.2 - 10.12.2023
Routing fixed mittels searchParametern und meta-Refresh-Weiterleitung von /de/hoga


## Website Update 0.1.1 - 09.12.2023
Komplettes Backend der Website überarbeitet. Nun bessere und schnellere Funktion und bessere Entwicklung möglich. Beim nächsten Update hoffentlich Versionskontrolle auf Github inkl. automatischem Deploy auf den Webspace


## Website Update 0.1.0 - 08.12.2023
Website ging online

src/pages/Hoga.js
const documents = [
    {
        name: "WI Unterlagen",
        link: "https://soulback.net/de/hogaDocs/WI_081223_comp.pdf",
        size: "3.357 kb",
        date: "08.12.2023"
    },
    {
        name: "LFK9 Unterlagen",
        link: "https://soulback.net/de/hogaDocs/LFK9_111223.pdf",
        size: "9.316 kb",
        date: "11.12.2023"
    }
];

export default function Hoga() {
    return(

        <main>
            <div className="mx-auto max-w-7xl py-6 sm:px-6 lg:px-8">
                <ul className="divide-y divide-gray-100 border-2 p-5">
                    {documents.map((doc) => (
                        <div className="flex flex-row py-5">
                            <div className="basis-1/4 p-2">
                                {doc.name}
                            </div>
                            <div className="basis-1/4 p-2 text-gray-400">
                                {doc.date}
                            </div>
                            <div className="basis-1/4 p-2 text-gray-400">
                                {doc.size}
                            </div>
                            <a href={doc.link}>
                                <div className="basis-1/4 bg-orange-400 hover:bg-orange-600 hover:border-gray-700 hover:text-gray-100 p-2 border-4">
                                    Download
                                </div>
                            </a>
                        </div>
                    ))}
                </ul>
            </div>
        </main>
    )
}

src/pages/dashboard.js
import { useEffect } from 'react';
import { useState } from 'react';
import Markdown from 'react-markdown';

const markdown = '# Hi, *Pluto*!';

export function Dashboard() {
    const [content, setContent] = useState("");

    useEffect(() => {
        fetch("./changelog.md")
            .then((res) => res.text())
            .then((text) => setContent(text));
    }, []);

    return(
        <div>
            <Markdown
                className="markdown"
                children={content}
                components={{
                    h1(props) {
                        const {node, ...rest} = props
                        return <h1 class='text-3xl font-bold tracking-tight text-gray-900' {...rest}/>
                    },
                    h2(props) {
                        const {node, ...rest} = props
                        return <h2 class='text-2xl font-bold tracking-tight text-gray-900' {...rest}/>
                    },
                    h3(props) {
                        const {node, ...rest} = props
                        return <h3 class='text-xl font-bold tracking-tight text-gray-900' {...rest}/>
                    },
                    ul(props){
                        const{node, ...rest} = props
                        return <ul class='list-disc' {...rest}/>
                    }
                }}
            / >
        </div>
    )
}

src/navigation/Navbar.js
import { Fragment, useEffect } from 'react'
import { Disclosure, Menu, Transition } from '@headlessui/react'
import { Bars3Icon, BellIcon, XMarkIcon } from '@heroicons/react/24/outline'
import { Link, Outlet, useNavigate } from 'react-router-dom'

const navigation = [
  { name: 'Dashboard', href: '/de', current: false },
  { name: 'HOGA Unterlagen', href: '/de/hoga', current: false }
]

function classNames(...classes) {
  return classes.filter(Boolean).join(' ')
}

const queryParams = new URLSearchParams(window.location.search);

export default function Navbar() {
  const navigate = useNavigate();
  const targetName = queryParams.get('targetname');

  useEffect(() => {
    if(targetName != null){
      if(targetName === 'hoga'){
        console.log("Setting Route to " + targetName);
        queryParams.set('targetname', null);
        navigate('/de/hoga');
      }
    }
  });

  return (
    <Disclosure as="nav" className="bg-gray-800">
      {({ open }) => (
        <>
          <div className="mx-auto max-w-7xl px-2 sm:px-6 lg:px-8">
            <div className="relative flex h-16 items-center justify-between">
              <div className="absolute inset-y-0 left-0 flex items-center sm:hidden">
                {/* Mobile menu button*/}
                <Disclosure.Button className="relative inline-flex items-center justify-center rounded-md p-2 text-gray-400 hover:bg-gray-700 hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white">
                  <span className="absolute -inset-0.5" />
                  <span className="sr-only">Open main menu</span>
                  {open ? (
                    <XMarkIcon className="block h-6 w-6" aria-hidden="true" />
                  ) : (
                    <Bars3Icon className="block h-6 w-6" aria-hidden="true" />
                  )}
                </Disclosure.Button>
              </div>
              <div className="flex flex-1 items-center justify-center sm:items-stretch sm:justify-start">
                <div className="flex flex-shrink-0 items-center">
                  <img
                    className="h-8 w-auto"
                    src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=500"
                    alt="Your Company"
                  />
                </div>
                <div className="hidden sm:ml-6 sm:block">
                  <div className="flex space-x-4">
                    {navigation.map((item) => (
                      <Link
                        key={item.name}
                        to={item.href}
                        className={classNames(
                          item.current ? 'bg-gray-900 text-white' : 'text-gray-300 hover:bg-gray-700 hover:text-white',
                          'rounded-md px-3 py-2 text-sm font-medium'
                        )}
                        aria-current={item.current ? 'page' : undefined}
                      >
                        {item.name}
                      </Link>
                    ))}
                  </div>
                </div>
              </div>
              <div className="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0">
                <button
                  type="button"
                  className="relative rounded-full bg-gray-800 p-1 text-gray-400 hover:text-white focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800"
                >
                  <span className="absolute -inset-1.5" />
                  <span className="sr-only">View notifications</span>
                  <BellIcon className="h-6 w-6" aria-hidden="true" />
                </button>

                {/* 
                Profile Dropdown
                <Menu as="div" className="relative ml-3">
                  <div>
                    <Menu.Button className="relative flex rounded-full bg-gray-800 text-sm focus:outline-none focus:ring-2 focus:ring-white focus:ring-offset-2 focus:ring-offset-gray-800">
                      <span className="absolute -inset-1.5" />
                      <span className="sr-only">Open user menu</span>
                      <img
                        className="h-8 w-8 rounded-full"
                        src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80"
                        alt=""
                      />
                    </Menu.Button>
                  </div>
                  <Transition
                    as={Fragment}
                    enter="transition ease-out duration-100"
                    enterFrom="transform opacity-0 scale-95"
                    enterTo="transform opacity-100 scale-100"
                    leave="transition ease-in duration-75"
                    leaveFrom="transform opacity-100 scale-100"
                    leaveTo="transform opacity-0 scale-95"
                  >
                  </Transition>
                </Menu>*/}
              </div>
            </div>
          </div>

          <Disclosure.Panel className="sm:hidden">
            <div className="space-y-1 px-2 pb-3 pt-2">
              {navigation.map((item) => (
                <Disclosure.Button
                  key={item.name}
                  as="a"
                  href={item.href}
                  className={classNames(
                    item.current ? 'bg-gray-900 text-white' : 'text-gray-300 hover:bg-gray-700 hover:text-white',
                    'block rounded-md px-3 py-2 text-base font-medium'
                  )}
                  aria-current={item.current ? 'page' : undefined}
                >
                  {item.name}
                </Disclosure.Button>
              ))}
            </div>
          </Disclosure.Panel>
        </>
      )}
    </Disclosure>
  )
}